baseapp_profiles/models.py

```
@@ -180,10 +184,254 @@ class ProfilableModel(models.Model):
+        return f"""
+            IF NEW.{self._profile_column} IS NULL THEN
+                WITH new_profile AS (
+                    INSERT INTO {profile_table} ({columns})
+                    VALUES ({values})
+                    ON CONFLICT (target_content_type_id, target_object_id) DO UPDATE SET modified = EXCLUDED.modified
+                    RETURNING id
+                )
+                UPDATE {self._self_table} SET {self._profile_column} = (SELECT id FROM new_profile) WHERE {self._pk} = NEW.{self._pk};
+            END IF;
+            RETURN NULL;
+        """
```

Ruff (0.15.10)

[error] Possible SQL injection vector through string-based query construction

(S608)